src/digest/base.cr

```diff
@@ -89,4 +88,28 @@ abstract class Digest::Base
 
     Base64.strict_encode(hashsum)
   end
+
+  def update(data) : self
+    update data.to_slice
+  end
+
+  def final : Bytes
+    dst = Bytes.new digest_size
+    final dst
+  end
+
+  # Dups and finishes the digest.
+  def digest : Bytes
+    dup.final
+  end
+
+  # Returns a hexadecimal-encoded digest.
+  def hexdigest : String
+    digest.hexstring
+  end
+
+  abstract def update(data : Bytes) : self
+  abstract def final(dst : Bytes) : Bytes
+  abstract def reset : self
+  abstract def digest_size : Int32
```

Comment:
Is this really useful as instance method or should it rather be a constant (or method) on the type.

Reply:
Can't have it that way when considering OpenSSL::Digest.  The `digest_size` is variable based on the passed in args.  Blake2 allows for variable length digests based on the initialization parameters.  So do many modern digests like Keccak.

Reply:
Okay. :+1: